File: Android V1.6/app/src/main/java/com/arcinteractive/spaces/data/model/SpaceNote.kt
package com.arcinteractive.spaces.data.model

import java.util.Date

data class SpaceNoteAttachment(val id: String, val name: String, val mimeType: String, val storagePath: String, val nonce: String, val isMedia: Boolean)
data class SpaceNote(
    val id: String, val spaceId: String, val title: String, val markdown: String,
    val attachments: List<SpaceNoteAttachment>, val links: List<SpaceLinkAttachment>,
    val createdBy: String, val createdAt: Date, val updatedAt: Date
)
data class SpaceNoteComment(
    val id: String, val noteId: String, val authorId: String, val authorName: String,
    val body: String, val createdAt: Date
)

File: iOS/Spaces V1.6/Models/SpaceNote.swift
import Foundation

struct SpaceNoteAttachment: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let mimeType: String
    let storagePath: String
    let nonce: String
    let isMedia: Bool
}

struct SpaceNote: Identifiable, Hashable {
    let id: String
    let spaceID: String
    var title: String
    var markdown: String
    var attachments: [SpaceNoteAttachment]
    var links: [SpaceLinkAttachment]
    let createdBy: String
    let createdAt: Date
    var updatedAt: Date
}

struct SpaceNoteComment: Identifiable, Hashable {
    let id: String
    let noteID: String
    let authorID: String
    let authorName: String
    let body: String
    let createdAt: Date
}

struct SpaceNotePayload: Codable {
    let title: String
    let markdown: String
    let attachments: [SpaceNoteAttachment]
    let links: [SpaceLinkAttachment]
}

struct SpaceNoteCommentPayload: Codable {
    let authorName: String
    let body: String
}
